### PR TITLE
Adds the Fenix Snackbar

### PR DESCRIPTION
**Pull request by boek**
Preview: https://cloudup.com/ctoYORuqqfw

### CODE REVIEW

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -132,4 +132,6 @@
     <color name="quick_action_pull_tab">#2915141A</color>
     <color name="quick_action_background_normal_theme">@color/toolbar_normal_theme</color>
     <color name="quick_action_background_private_theme">@color/toolbar_dark_mode</color>
+
+    <color name="fenix_snackbar_background">#232749</color>
```
Comment (sblatz) [inline app/src/main/res/values/colors.xml]: Do we want this to look the same in all themes?

File: app/src/main/java/org/mozilla/fenix/utils/ItsNotBrokenSnack.kt [hunk 2]
```diff
@@ -6,32 +6,25 @@ package org.mozilla.fenix.utils
...
-                    )
-                    show()
-                }
-        }
+            context.asActivity()?.window?.decorView?.findViewById<View>(android.R.id.content) as ViewGroup
+
+        FenixSnackbar.make(rootView, Snackbar.LENGTH_SHORT)
+            .setText(message.replace("%", issueNumber))
+            .setAction("Add Tab to Issue") {
+                context.components.useCases.tabsUseCases.addTab
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/utils/ItsNotBrokenSnack.kt]: The way this works right now, we're opening a tab in the background. Do we want to instead use `openToBrowserAndLoad`?

File: app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        fun make(parent: ViewGroup, duration: Int): FenixSnackbar {
+            val inflater = LayoutInflater.from(parent.context)
+            val content = inflater.inflate(R.layout.fenix_snackbar, parent, false)
+
+            // create snackbar with custom view
+            val callback = FenixSnackbarCallback(content)
+            val fenixSnackbar = FenixSnackbar(parent, content, callback)
+
+            // set snackbar duration
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt]: Does this need a comment? Seems pretty obvious from the line below

File: app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        private const val minTextSize = 12
+        private const val maxTextSize = 18
+        private const val increaseTouchableAreaBy = 16
+        private const val stepGranularity = 1
+
+        fun make(parent: ViewGroup, duration: Int): FenixSnackbar {
+            val inflater = LayoutInflater.from(parent.context)
+            val content = inflater.inflate(R.layout.fenix_snackbar, parent, false)
+
+            // create snackbar with custom view
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt]: See comment below, but if we want to keep it: 

very picky nit: we should capitalize the comment 😛

File: app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+class FenixSnackbar private constructor(
+    parent: ViewGroup,
+    content: View,
+    contentViewCallback: FenixSnackbarCallback
+) : BaseTransientBottomBar<FenixSnackbar>(parent, content, contentViewCallback) {
+
+    init {
+        view.background = null
+        content.snackbar_btn.increaseTapArea(increaseTouchableAreaBy)
```
Comment (sblatz) [inline app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt]: nit: the rest of our `increaseTapAreas` pass params with names like: `closeButtonIncreaseDps`. Perhaps we should mirror this for consistency?

File: app/src/main/java/org/mozilla/fenix/utils/ItsNotBrokenSnack.kt (hunk 2 above)
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/utils/ItsNotBrokenSnack.kt]: I didn't want to change the current behavior in this PR

File: app/src/main/res/values/colors.xml (hunk 1 above)
Comment (boek) [reply app/src/main/res/values/colors.xml]: Good question. I'll ask @brampitoyo and follow up with a new PR :+1:

**Pull request merged**